src/settings.rs

```diff
@@ -84,6 +91,7 @@ impl Default for Settings {
             profile: Some("default".to_owned()),
             overrides: BTreeMap::new(),
             pgp_keys: None,
+            auto_self_update: Some(SelfUpdateMode::Enable),
```

Comment:
Either this should default to None, or the type should not have Option<> in it.

If we want to support writing the file back with an unset key unset (which I think is probably good), then we need to keep None as the default value and only cast up to Enable on queries. Otherwise new files will be written with a value in them.

And get_self_update_mode already returns Enable when the value is None....